### PR TITLE
Remove deprecated installation script

### PR DESCRIPTION
**Pull request by Secrus**
# Pull Request Check List

Resolves: #6676

Scheduled removal of the old, unmaintained installer script.

### CODE REVIEW

File: docs/_index.md [hunk 1]
```diff
@@ -46,13 +46,8 @@ The script can be executed directly (i.e. 'curl python') or downloaded and then
...
 {{% warning %}}
-The `get-poetry.py` installer has been deprecated and removed. If you installed
-Poetry using `get-poetry.py`, please uninstall (as documented in the relevant
-step below), and then follow these installation instructions.
-
-The previous `install-poetry.py` script as included in the Poetry repository
-is deprecated and frozen. Please migrate to the standalone version described
-above, as the in-tree version is [scheduled to be removed](https://github.com/python-poetry/poetry/issues/6676).
+The `install-poetry.py` installer has been deprecated and removed.
+Please migrate to the standalone version described above.
```
Comment (radoering) [inline docs/_index.md:50]:
```suggestion
The `install-poetry.py` installer has been deprecated and removed from the Poetry repository.
Please migrate from the in-tree version to the standalone version described above.
```

**Review by radoering — CHANGES_REQUESTED** [review]
`install-poetry.py` is mentioned in two places in `_index.md` (pre-release versions and CI recommendations) that need to be changed.

File: docs/_index.md (hunk 1 above)
Comment (Secrus) [reply docs/_index.md]: > `install-poetry.py` is mentioned in two more places in `_index.md` (pre-release versions and CI recommendations) that need to be changed.

I have updated the pre-release part. As to CI recommendations, it states clearly to download `install-poetry.py` from the `install.python-poetry.org` repo, so I would leave it as it is unless you have some idea how to rephrase it.

Comment (radoering) [reply docs/_index.md]: I did not read the details. Should be fine.

File: docs/_index.md [hunk 2]
```diff
@@ -92,7 +87,7 @@ If you wish to change this, you may define the `$POETRY_HOME` environment variab
 curl -sSL https://install.python-poetry.org | POETRY_HOME=/etc/poetry python3 -
 ```
 
-If you want to install prerelease versions, you can do so by passing `--preview` option to `install-poetry.py`
+If you want to install prerelease versions, you can do so by passing `--preview` option to installation script
```
Comment (radoering) [inline docs/_index.md:90]:
```suggestion
If you want to install prerelease versions, you can do so by passing the `--preview` option to the installation script
```

**Review by radoering — APPROVED** [review]

**Pull request merged**